src/Functions/sqid.cpp

```diff
@@ -124,7 +124,7 @@ class FunctionSqidDecode : public IFunction
                 std::string_view sqid = col_non_const->getDataAt(i).toView();
                 std::vector<UInt64> integers = sqids.decode(String(sqid));
                 res_nested_data.insert(integers.begin(), integers.end());
-                res_offsets_data.push_back(integers.size());
+                res_offsets_data.push_back(i == 0 ? integers.size() : res_offsets_data.back() + integers.size());
```

Comment:
`res_offsets_data` is `PaddedPODArray` and `back()` returns 0 when it's empty, so no need to check for `i = 0`

Reply:
Tnx, fixed.